blocks/subscriptions-block/components/usePaywall.js

```diff
@@ -57,7 +57,7 @@ const usePaywall = () => {
 			!isPaywalled &&
```

Comment:
The message above is correct, `rules` is only used inside of the `useEffect` below and should be moved into it. The you don't need the `!isServerSide()` check anymore, since `useEffect` only runs on the client side render.